porch/func/internal/evaluator.go

```diff
@@ -32,6 +33,43 @@ import (
 type evaluator struct {
 	pb.UnimplementedFunctionEvaluatorServer
 
+	builtin *builtinEvaluator
+	exec    *executableEvaluator
+}
+
+func NewEvaluatorWithConfig(functions string, config string) (pb.FunctionEvaluatorServer, error) {
+	builtinEval, err := newBuiltInEvaluator()
+	if err != nil {
+		return nil, err
+	}
+	execEval, err := newExecutableEvaluatorWithConfig(functions, config)
+	if err != nil {
+		return nil, err
+	}
+	return &evaluator{
+		builtin: builtinEval,
+		exec:    execEval,
+	}, nil
+}
+
+func (e *evaluator) EvaluateFunction(ctx context.Context, req *pb.EvaluateFunctionRequest) (*pb.EvaluateFunctionResponse, error) {
+	// try to evaluate a KRM function in the following order:
+	// 1) builtin 2) executable 3) pod evaluator (not implemented yet)
+	resp, err := e.builtin.EvaluateFunction(ctx, req)
+	var ufe UnsupportedFunctionError
+	if !errors.As(err, &ufe) {
+		return nil, err
+	}
+	resp, err = e.exec.EvaluateFunction(ctx, req)
+	if errors.As(err, &ufe) {
+		return nil, status.Errorf(codes.NotFound, ufe.Error())
+	}
+	return resp, err
+}
+
+type executableEvaluator struct {
+	pb.UnimplementedFunctionEvaluatorServer
```

Comment:
We should not need this line anymore since the `evaluator` struct above implements the gRPC interface and we don't expose the `executableEvaluator` as gRPC (it is delegated to by the `evaluator`